src/codegeneration/ES6ClassTransformer.js

```diff
@@ -0,0 +1,265 @@
+// Copyright 2015 Traceur Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the 'License');
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an 'AS IS' BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import {
+  CONSTRUCTOR
+} from '../syntax/PredefinedName.js';
+import {
+  AnonBlock,
+  ClassExpression,
+  ClassDeclaration,
+  GetAccessor,
+  PropertyMethodAssignment,
+  SetAccessor,
+  LiteralPropertyName,
+} from '../syntax/trees/ParseTrees.js';
+import {
+  CALL_EXPRESSION,
+  EXPRESSION_STATEMENT,
+  GET_ACCESSOR,
+  PROPERTY_METHOD_ASSIGNMENT,
+  PROPERTY_VARIABLE_DECLARATION,
+  SET_ACCESSOR,
+  SUPER_EXPRESSION
+} from '../syntax/trees/ParseTreeType.js';
+import {TempVarTransformer} from './TempVarTransformer.js';
+import {ParseTreeTransformer} from './ParseTreeTransformer.js';
+import {
+  createBindingIdentifier,
+  createCommaExpression,
+  createEmptyParameterList,
+  createExpressionStatement,
+  createFunctionBody,
+  createIdentifierToken,
+  createImmediatelyInvokedFunctionExpression,
+  createPropertyNameAssignment,
+  createParenExpression,
+} from './ParseTreeFactory.js';
+import {
+  parseExpression,
+  parsePropertyDefinition,
+  parseStatement,
+} from './PlaceholderParser.js';
+import {propName} from '../staticsemantics/PropName.js';
+import {prependStatements} from './PrependStatements.js';
+
+// TODO(vicb): comment
+
+export class ES6ClassTransformer extends TempVarTransformer {
+  /**
+   * @param {UniqueIdentifierGenerator} identifierGenerator
+   */
+  constructor(identifierGenerator) {
+    super(identifierGenerator);
+  }
+
+  transformClassElements_(tree) {
+    let superClass = this.transformAny(tree.superClass);
+
+    let elements = [];
+    let initInstanceVars = [], initStaticVars = [];
+    let constructor;
+    let constructorIndex = 0;
+
+    tree.elements.forEach((tree) => {
+      let initVars;
+      if (tree.isStatic) {
+        initVars = initStaticVars;
+      } else {
+        initVars = initInstanceVars;
+      }
+
+      switch (tree.type) {
+        case GET_ACCESSOR:
```

Comment:
This can be written as:

```js
case GET_ACCESSOR:
case SET_ACCESSOR:
  elements.push(this.transformAny(tree));
  break;
```

There is no need to create that extra tree here.